src/walk_dirs.rs

```diff
@@ -57,26 +57,36 @@ lazy_static! {
 pub fn discover_fences_and_files(
     start_path: &str,
     ignore_external_fences: ExternalFences,
+    ignored_dirs: Vec<regex::Regex>,
 ) -> Vec<WalkFileData> {
     let walk_dir = WalkDirGeneric::<(TagList, WalkFileData)>::new(start_path).process_read_dir(
         move |read_dir_state, children| {
             // Custom filter -- retain only directories and fence.json files
             children.retain(|dir_entry_result| {
+                match dir_entry_result {
+                    Ok(dir_entry) => {
+                        if let Some(slashed) = dir_entry.path().to_slash() {
+                            return !(ignore_external_fences == ExternalFences::Ignore
+                                && slashed.to_string().ends_with("/node_modules"))
+                                && !ignored_dirs.iter().any(|d| {
+                                    d.is_match(&slashed)
```

Comment:
This behaviour is incorrect. Join the filename onto the slashed path, then check the regex against the resulting path.

As implemented now, `^lib` will incorrectly match `packages/lib`
And
`packages/fence.json` will fail to match `packages/fence.json`

Its also worth adding tests for those examples.

Reply:
Seems that at this point `slashed` is either a dir or a file, so comparing just `slashed` should be enough.

Reply:
Oh awesome :+1: